meta_test.go

```diff
@@ -1,216 +1,223 @@
+	return shutil.CopyFile(filepath.Join(mockDir, metaFile), filepath.Join(testDir, metaFile), false)
+}
 
-	stdout = new(bytes.Buffer)
-	getMeta("obj.momo", mockDir, testFile, stdout, false, nil)
-	expected = []byte("{\"toke\":\"toke\"}")
-	if bytes.Compare(expected, stdout.Bytes()) != 0 {
-		t.Fatalf("not matched. expected '%v', actual '%v'", string(expected), string(stdout.Bytes()))
-	}
+func (s *MetaSuite) TestSetupDir() {
+	var err error
+	var data []byte
 
-	stdout = new(bytes.Buffer)
-	getMeta("ary", mockDir, testFile, stdout, false, nil)
-	expected = []byte("[\"aaa\",\"bbb\",{\"ccc\":{\"ddd\":[1234567,2,3]}}]")
-	if bytes.Compare(expected, stdout.Bytes()) != 0 {
-		t.Fatalf("not matched. expected '%v', actual '%v'", string(expected), string(stdout.Bytes()))
-	}
+	_, err = os.Stat(testFilePath)
+	Assert := s.Assert()
+	Assert.NoError(err, "could not create %s in %s", testFilePath, testDir)
 
-	stdout = new(bytes.Buffer)
-	getMeta("ary[0]", mockDir, testFile, stdout, false, nil)
-	expected = []byte("aaa")
-	if bytes.Compare(expected, stdout.Bytes()) != 0 {
-		t.Fatalf("not matched. expected '%v', actual '%v'", string(expected), string(stdout.Bytes()))
-	}
+	data, err = ioutil.ReadFile(testFilePath)
+	Assert.NoError(err, "could not read %s in %s", testFilePath, testDir)
+	Assert.Equal("{}", string(data), "%s does not have an empty JSON object: %v", testFilePath, string(data))
+}
 
-	stdout = new(bytes.Buffer)
-	getMeta("ary[2]", mockDir, testFile, stdout, false, nil)
-	expected = []byte("{\"ccc\":{\"ddd\":[1234567,2,3]}}")
-	if bytes.Compare(expected, stdout.Bytes()) != 0 {
-		t.Fatalf("not matched. expected '%v', actual '%v'", string(expected), string(stdout.Bytes()))
-	}
+func (s *MetaSuite) TestExternalMetaFile() {
+	s.MetaSpec.MetaFile = externalFile
+	_, _ = s.MetaSpec.SetupDir()
+	_ = os.Remove(externalFilePath)
 
-	stdout = new(bytes.Buffer)
-	getMeta("ary[2].ccc", mockDir, testFile, stdout, false, nil)
-	expected = []byte("{\"ddd\":[1234567,2,3]}")
-	if bytes.Compare(expected, stdout.Bytes()) != 0 {
-		t.Fatalf("not matched. expected '%v', actual '%v'", string(expected), string(stdout.Bytes()))
-	}
+	// Test set (meta file is not meta.json, should fail)
+	err := s.MetaSpec.Set("str", "val")
+	Require := s.Require()
+	Require.Error(err, "error should be occured")
 
-	stdout = new(bytes.Buffer)
-	getMeta("ary[2].ccc.ddd", mockDir, testFile, stdout, false, nil)
-	expected = []byte("[1234567,2,3]")
-	if bytes.Compare(expected, stdout.Bytes()) != 0 {
-		t.Fatalf("not matched. expected '%v', actual '%v'", string(expected), string(stdout.Bytes()))
-	}
+	// Test get
+	Require.NoError(s.CopyMockFile(externalFile))
 
-	stdout = new(bytes.Buffer)
-	getMeta("ary[2].ccc.ddd[1]", mockDir, testFile, stdout, false, nil)
-	expected = []byte("2")
-	if bytes.Compare(expected, stdout.Bytes()) != 0 {
-		t.Fatalf("not matched. expected '%v', actual '%v'", string(expected), string(stdout.Bytes()))
-	}
+	got, err := s.MetaSpec.Get("str")
```

Comment:
`staticcheck`: `meta_test.go:96:7: this value of err is never used (SA4006)` - should the test check if `err` is not nil?

Reply:
Done